File: scripts/Intersection of Two Linked Lists/Intersection of Two Linked Lists.py
class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> Optional[ListNode]:
        m = 0
        n = 0
        temp = headA
        while temp != None:
            m+=1
            temp = temp.next
        temp = headB
        while temp != None:
            n+=1
            temp = temp.next
        diff = 0
        if m>=n : 
            diff = m-n
        else:
            diff = n-m
        p1 = headA
        p2 = headB
        if max(m,n) == m:
            while diff > 0:
                p1 = p1.next
                diff-=1
        else:
            while diff > 0:
                p2 = p2.next
                diff-=1
        while p1 != None and p2!=None:
            if p1 == p2:
                return p1
            p1 = p1.next
            p2 = p2.next
        return None

File: scripts/algorithms/I/Intersection of Two Linked Lists/Intersection of Two Linked Lists.js
// Runtime: 134 ms (Top 54.42%) | Memory: 52.1 MB (Top 6.15%)

var getIntersectionNode = function(headA, headB) {
   let node1 = headA;
   let node2 = headB;

    const set = new Set()

    while(node1 !== undefined || node2!== undefined){
        if(node1 && set.has(node1)){
            return node1
        }
        if(node2 && set.has(node2)){
            return node2
        }
        if(node1){
            set.add(node1)
            node1 = node1.next;
        }
        else if(node2){
            set.add(node2)
            node2 = node2.next;
        }
        else {return}
    }
 };